### PR TITLE
Add support for @typeschema/class-validator

### PR DESCRIPTION
**Pull request by niemyjski**
Added support and tests for https://github.com/decs/typeschema/tree/main/packages/class-validator

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -105,6 +105,8 @@
 		"@exodus/schemasafe": "^1.3.0",
 		"@sinclair/typebox": ">=0.32.30 <1",
 		"@sveltejs/kit": "1.x || 2.x",
+		"@typeschema/class-validator": "^0.1.2",
+		"class-validator": "^0.13.1",
```
Comment (niemyjski) [inline package.json:109]: I didn't think this should be needed as there should be a transitive package reference but tests failed without it.

File: src/lib/adapters/class-validator.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	type ValidationResult,
+	type ClientValidationAdapter,
+	type RequiredDefaultsOptions,
+} from './adapters.js';
+import { memoize } from '$lib/memoize.js';
+
+import {
+	validate as classValidatorValidate,
+	type Infer,
+	type InferIn,
```
Comment (niemyjski) [inline src/lib/adapters/class-validator.ts:14]: This would only compile if I used this Infer and not the one from adapters.

File: tsconfig.json [hunk 3]
```diff
@@ -4,6 +4,8 @@
 		"allowJs": true,
 		"checkJs": true,
 		"esModuleInterop": true,
+		"emitDecoratorMetadata": true,
+		"experimentalDecorators": true,
```
Comment (niemyjski) [inline tsconfig.json:8]: This was needed for the class-validator test decorators.

**Review by niemyjski — COMMENTED** [review]
Self reviewed

**Comment by ciscoheat** [comment]
You also need to add the Infer type for the library in `src\lib\adapters\typeSchema.ts` (that could handle some of the typing problems you encountered)

**Pull request merged**